docs: SQLair tutorial updates (#165)
canonical/sqlair: docs/tutorial/new-tutorial.md

(tutorial)=
# SQLair tutorial

Imagine you are writing a Go program that handles your customers and their
orders. The business logic is contained in the Go program but the customer and
order data is stored in a SQL database. In this tutorial, we'll see how easy
SQLair makes it to map the database between the database and the Go program.
Let's get started!

```{note}
This tutorial will take around 30 minutes to complete. If you want a quicker
start see {ref}`prepare-and-run`.
```

> ## Setting up the environment

Reviewer: I'd rephrase as "Set up your test environment"

> To start with create a new folder called tutorial and initialise a new Go project.

Reviewer: ```suggestion
To start with, create a new folder called `tutorial` and initialise a new Go project.
```

Reviewer: In fact, I'd simplify even further:

```suggestion
The first thing you'll need is a Go project. Let's create one:
```

Then I'd add line by line comments in the code block, just to help someone who's really clueless catch up.

```bash
mkdir tutorial 
cd tutorial
go mod init sqlair-tutorial
```

We first need to set up the backbone of our project, the database. The easiest
way to create a SQL database in Go is to use [SQLite](https://www.sqlite.org/)
with the [Go driver from mattn](https://github.com/mattn/go-sqlite3) though
SQLair works with any database driver that can interface with
[database/sql](https://pkg.go.dev/database/sql). With SQLite, we can create an in
process, in memory database that can be populated within the program and that
will disappear when the program finishes running.

We also want to set the database up with our customer and order schema. We will
have two tables. One for customers and the other for orders.

To create the database, create a new `main.go` file in the project, open it in

> your favourite editor and copy in the code below.

Reviewer: Again, not sure how to do multi-line suggestions, but here what's I'd do: 

```suggestion
To create the database, in your Go project create a `main.go` file; open it in your favourite editor; then copy in the code below:
```
